### ISSUE TITLE
Schrodinger File Permission? permissions in `/mnt/` drive keeps changing randomly when observed

### ISSUE THREAD

**Issue by LearnedVector** [microsoft/WSL#5463]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.20150.1000]
Your Distribution version: Ubuntu 20.04
Whether the issue is on WSL 2 and/or WSL 1: WSL2
```

# Steps to reproduce

The file permissions keep changing randomly causing me to not be able to access files.

I have a second drive in `/mnt/` that is NTFS formatted and some data files I need to be able to read and write. I have followed the guide to set file permissions found on the [Microsoft documentation site](https://docs.microsoft.com/en-us/windows/wsl/file-permissions).

in `/etc/wsl.conf` I have the follow configuration
```
[automount]
enabled = true
root = /mnt/
options = "metadata,umask=22,fmask=11"
```

Most of my files have these permissions
```
---------- 1 warmachine warmachine 30764 Jun 20 02:21 0.wav
---------- 1 warmachine warmachine 30764 Jun 20 02:21 1.wav
---------- 1 warmachine warmachine 30764 Jun 20 02:21 10.wav
...
```

I used chmod to change file permission to...
```
-rwxr--r-- 1 warmachine warmachine 30764 Jun 20 02:21 0.wav
-rwxr--r-- 1 warmachine warmachine 30764 Jun 20 02:21 1.wav
-rwxr--r-- 1 warmachine warmachine 30764 Jun 20 02:21 10.wav
...
```

my issue is that I have some sort of weird [Schrodinger cat](https://en.wikipedia.org/wiki/Schr%C3%B6dinger%27s_cat#:~:text=Schr%C3%B6dinger's%20cat%20is%20a%20thought,mechanics%20applied%20to%20everyday%20objects.) situation on my file permissions. The files would randomly switch back to this format
```
---------- 1 warmachine warmachine 30764 Jun 20 02:21 0.wav
```

but if I `ls -l` the files a few times, it would then randomly revert back to chmod permissions
```
-rwxr--r-- 1 warmachine warmachine 30764 Jun 20 02:21 0.wav
```

but then after a while the file would revert back to permission before I chmod.

This is an issue because when i try to access my files programmatically i get a permission denied error. Am I doing something wrong? any way to fix this Schrodinger file permission?

#  Expected behavior

The files should maintain permission from chmod without randomly changing.

# Actual behavior

TLDR: file permissions randomly changing state in the `/mnt/` drive causing permission denied issue when trying to access it programmatically.

**Comment by rlabrecque** [comment]
I am seeing something similar, but I'm seeing it happen everywhere, not just /mnt/ (though I believe I had seen this on /mnt/ prior to ditching it though.)

Windows 10 10.0.19041 Build 19041
Debian 10.4 Buster
WSL2

My wsl.conf:
```
[automount]
options= "metadata"
```

I'm accessing files in `\\wsl$\Debian\home\riley` and my shell scripts keep reverting the executable bit, seemingly randomly. They will run fine, then 5 minutes later without touching them they will revert to `-x`.

**Comment by tigerjack** [comment]
Same observed behaviour here.
WSL2 running Arch Linux, Windows 10 2004 with build 20262.1, no `wsl.conf` file used.
This is particularly annoying since I have to manually run a `chmod -R 777 *` for all the files I want to access from the linux system.
